internal/handlers/images.go

```diff
@@ -1,34 +1,45 @@
 package handlers
 
 import (
+	"bytes"
+	"compress/gzip"
 	"fmt"
+	"io"
 	"net/http"
+	"net/url"
 	"regexp"
+	"time"
 
+	"github.com/cavaliercoder/go-cpio"
 	"github.com/openshift/assisted-image-service/pkg/imagestore"
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 	metrics "github.com/slok/go-http-metrics/metrics/prometheus"
 	"github.com/slok/go-http-metrics/middleware"
 	stdmiddleware "github.com/slok/go-http-metrics/middleware/std"
 )
 
 type ImageHandler struct {
-	ImageStore imagestore.ImageStore
+	ImageStore            imagestore.ImageStore
+	AssistedServiceScheme string
+	AssistedServiceHost   string
+	GenerateImageStream   isoeditor.StreamGeneratorFunc
 }
 
 var _ http.Handler = &ImageHandler{}
 
-var clusterRegexp = regexp.MustCompile(`/images/.+`)
+var clusterRegexp = regexp.MustCompile(`^/images/(.+)`)
```

Comment:
Nice change.

Reply:
This was actually breaking things so I created a separate commit and tests for this.